backend/api/views.py

```diff
@@ -1,3 +1,19 @@
-from django.shortcuts import render
+from rest_framework import viewsets, mixins
+from serializers import BusinessSerializer
+
+from sodapy import Socrata
+
+class BusinessesViewSet(mixins.ListModelMixin,
+                        viewsets.GenericViewSet):
+
+    serializer_class = BusinessSerializer
+
+    def get_queryset(self):
+        client = Socrata("data.cityofnewyork.us", "z9bHXnSMLOVBFTGrELvWCcCBN")
+        businesses = client.get("9w7m-hzhe", content_type="json",
+                                limit="500000",
+                                select="camis, dba, boro, building, street, zipcode, cuisine_description")
```

Comment:
Since you're doing the 3rd-party requests as part of the critical path (i.e. your user doesn't get data until *you* get data in turn from `Socrata`) you definitely want to be parsimonious in the amount of data you're retrieving as well as the number of calls you're making. The ideal would be to make one `Socrata` call for the absolute minimum of data that you need.

In this case you're forming the response to a bare `/businesses` call from the FE with no query string params. Think about what you're trying to show to the user in this case and then try and pull the minimum amount of data for that. For example, if the goal is to show a page of the 10 most recently updated businesses then that is what you should pull from `Socrata` and no more than that. I can't think of any valid use case where you would need 500,000 records.